wasm_src/ast_wrapper/ast_wrapper.cc

```diff
@@ -535,6 +535,29 @@ EMSCRIPTEN_KEEPALIVE int transform3D(AstSpecFrame* wcsinfo, double x, double y,
     return 0;
 }
 
+EMSCRIPTEN_KEEPALIVE int transform3DArray(AstFrameSet* wcsinfo, int npoint, double x[], double y[], double z[], const int forward, double out[])
+{
+    if (!wcsinfo)
+    {
+        return 1;
+    }
+
+    double in[3][npoint];
```

Comment:
@YuHsuan-Hwang surely this is not standard C++? We're allocating `3 * npoint` doubles in the stack. Since `npoint` is variable, this shouldn't work :thinking:  I know it's allowed in C99, but not in C++. The correct way to do this would be to allocate the array using malloc on the JS side. Alternatively, you could allocate a single input array of size `npoint * 3` and then just point x, y and z to the appropriate memory locations within?

Reply:
fixed.